src/mono/System.Private.CoreLib/System.Private.CoreLib.csproj

```diff
@@ -189,7 +189,7 @@
       <Compile Include="$(BclSourcesRoot)\System\TypeLoadException.Mono.cs" />
       <Compile Include="$(BclSourcesRoot)\System\ValueType.cs" />
       <Compile Include="$(BclSourcesRoot)\System\__ComObject.cs" />
-      <Compile Include="$(BclSourcesRoot)\System\Collections\Generic\ArraySortHelper.Mono.cs" />
+      <Compile Include="$(BclSourcesRoot)\System\Collections\Generic\ArraySortHelper.Mono.cs" Condition="'$(UnityAot)' != 'true'" />
```

Comment:
I think you can move this to `unity-aot.props` as well.  Try using something like
```
Compile Exclude="$(BclSourcesRoot)\System\Collections\Generic\ArraySortHelper.Mono.cs"
```